download-directory/Jenkinsfile

```diff
@@ -0,0 +1,78 @@
+import groovy.json.JsonSlurper
+// This Jenkinsfile is used by Jenkins to run the DownloadDirectory step of Reactome's release.
+// It requires that the AddLinks-Insertion step has been run successfully before it can be run.
+pipeline {
+	agent any
+
+	stages {
+		// This stage checks that an upstream project, AddLinks-Insertion, was run successfully for its last build.
+		stage('Check AddLinks-Insertion build succeeded'){
+			steps{
+				script{
+					def currentRelease = (pwd() =~ /Releases\/(\d+)\//)[0][1];
+					// This queries the Jenkins API to confirm that the most recent build of AddLinks-Insertion was successful.
+					def addLinksInsertionStatusUrl = httpRequest authentication: 'jenkinsKey', validResponseCodes: '200:404', url: "${env.JENKINS_JOB_URL}/job/$currentRelease/job/AddLinks-Insertion/lastBuild/api/json"
```

Comment:
These valid response codes look to be repeated in the PR.  Can they be consolidated to a single place?  Also, would other 2XX and 3XX codes be considered valid or only 200?

Reply:
The validResponseCodes are actually a property used by this httpRequest library, and it is actually meant to indicate the range of possible response codes that will be accepted. The default behaviour allowed for, I think, included all 2XX and 3XX ones, but not the 404 (why, is beyond me), which is why I needed to actually specify the acceptable range

Reply:
Regarding consolidation, I believe they can be incorporated into an environmental variable

Reply:
These have been changed to environmental variables